code/modules/antagonists/abductor/equipment/glands/heal.dm

```diff
@@ -0,0 +1,177 @@
+/obj/item/organ/heart/gland/heal
+	true_name = "organic replicator"
+	cooldown_low = 200
+	cooldown_high = 400
+	uses = -1
+	human_only = TRUE
+	icon_state = "health"
+	mind_control_uses = 3
+	mind_control_duration = 3000
+
+/obj/item/organ/heart/gland/heal/activate()
+	if(!MOB_ORGANIC in owner.mob_biotypes)
+		return
+
+	for(var/organ in internal_organs)
+		if(istype(organ, /obj/item/organ/cyberimp))
```

Comment:
This won't eject stuff like cybernetic organs, only implants.

Something more general like this might be neat.
```
var/organ_rejected = FALSE
for(var/obj/item/organ/o in internal_organs)
	if(o.status == ORGAN_ROBOTIC))
		owner.visible_message("<span class='warning'>[owner] vomits up his [o]!</span>", "<span class='userdanger'>You suddenly vomit up your [o]!</span>")
		owner.vomit(0, TRUE, TRUE, 1, FALSE, FALSE, FALSE, TRUE)
		o.Remove(owner)
		o.forceMove(owner.drop_location())
		organ_rejected |= TRUE
if(organ_rejected)
	owner.dna.species.regenerate_organs(owner, replace_current=FALSE)
```

Reply:
cyber organs are handled in each organ, although the ORGAN_ROBOTIC check is cleaner than the current type check